docs/examples/Ross-Macdonald.jl

```diff
@@ -0,0 +1,185 @@
+
+dZdt = function(u,x,p,t)
+    Z = u[1]
+    X = x[1]
+    [p.a*p.c*X*(exp(-p.g*p.n) - Z) - p.g*Z]
+end
+
+dXdt = function(u,x,p,t)
+    X = u[1]
+    Z = x[1]
+    [p.m*p.a*p.b*Z*(1 - X) - p.r*X]
+end
+    
+#- 
+
+mosquito_model = ContinuousMachine{Float64}(1, 1, 1, dZdt, (u,p,t) -> u)
+human_model = ContinuousMachine{Float64}(1, 1, 1, dXdt, (u,p,t) ->  u)
+
+malaria_model = oapply(bloodmeal, 
+    Dict(:humans => human_model, :mosquitos => mosquito_model)
+)
+
+#- 
+params = LVector(a = 0.3, b = 0.55, c = 0.15, 
+    g = 0.1, n = 10, r = 1.0/200, m = 0.5)
+
+u0 = [0.1, 0.3]
+tspan = (0.0, 365.0*2)
+
+prob = ODEProblem(malaria_model, u0, tspan, params)
+sol = solve(prob, Tsit5());
+#- 
+
+plot(
+    sol, lw=2,
+    title = "Ross-Macdonald Malaria model", 
+    label=["mosquitos" "humans"], 
+    xlabel = "Time", ylabel = "Proportion infectious"
+)
+
+## Plot the equilibrium behavior as well
+a, b, c, g, n, r, m = params
+R0 = (m*a^2*b*c*exp(-g*n))/(r*g)
+X̄ = (R0 - 1)/(R0 + (a*c)/g)
+Z̄ = (a*c*X̄)/(g + a*c*X̄)*exp(-g*n)
+
+N = length(sol)
+plot!(sol.t, fill(X̄, N), label = "human equilibrium", ls = :dash, lw = 2)
+plot!(sol.t, fill(Z̄, N), label = "mosquito equilibrium", ls = :dash, lw = 2)
+
+# ## Delay Model 
+# The previous models did not capture the incubation period for the disease in the
+# mosquito population. To do so we can replace the models with delay differential equations 
+# and apply them to the same diagram of systems representing the bloodmeal.
+
+dzdt_delay = function(u,x,h,p,t)
+    Y, Z = u
+    Y_delay, Z_delay = h(p, t - p.n)
+    X, X_delay = x[1]
+    
+    [p.a*p.c*X*(1 - Y - Z) -
+        p.a*p.c*X_delay*(1 - Y_delay - Z_delay)*exp(-p.g*p.n) -
+        p.g*Y,
+    p.a*p.c*X_delay*(1 - Y_delay - Z_delay)*exp(-p.g*p.n) -
+        p.g*Z]
+end
+
+dxdt_delay = function(u,x,h,p,t)
+    X, = u
+    Z, _ = x[1]
+    [p.m*p.a*p.b*Z*(1 - X) - p.r*X]
+end
+
+#- 
+
+
+mosquito_delay_model = DelayMachine{Float64, 2}(
+    1, 2, 1, dzdt_delay, (u,h,p,t) -> [[u[2], h(p,t - p.n)[2]]])
+
+human_delay_model = DelayMachine{Float64, 2}(
+    1, 1, 1, dxdt_delay, (u,h,p,t) -> [[u[1], h(p, t - p.n)[1]]])
+
+malaria_delay_model = oapply(bloodmeal, 
+    Dict(:humans => human_delay_model, :mosquitos => mosquito_delay_model)
+)
+#- 
+params = LVector(a = 0.3, b = 0.55, c = 0.15, 
+    g = 0.1, n = 10, r = 1.0/200, m = 0.5)
+
+u0_delay = [0.09, .01, 0.3]
+tspan = (0.0, 365.0*3)
+hist(p,t) = u0_delay;
+
+prob = DDEProblem(malaria_delay_model, u0_delay, [], hist, tspan, params)
+alg = MethodOfSteps(Tsit5())
+sol = solve(prob, alg)
+
+plot(
+    sol, lw=2,
+    title = "Ross-Macdonald malaria model",
+    label=["non-infectious mosquito population" "infectious mosquito population" "host population"],
+    xlabel = "Time", ylabel = "Proportion infectious"
+)
+
+# Notice that while the limiting behavior of the ODE and DDE models are identical, they may have 
```

Comment:
@slwu89 Would love your review of this sentence!

Reply:
@slibkind I hope this is useful, or at least helps inspire a better sentence (or two)!
"While the equilibrium points of the two models are identical, they exhibit different dynamical behavior before settling down to equilibrium. Because models are often used to examine how the system may respond to intervention, incorporating additional biological realism can produce more plausible results regarding the expected time for interventions to affect the system. In any case, such simple models are best used as tools to explain basic concepts in malaria epidemiology rather than descriptions of real systems, which are likely far from equilibrium and affected by weather, climate, policy, and other external forces."

Reply:
Perfect! Thank you so much.